### PR TITLE
PYTHON-1915 Prohibit copying ClientSession objects

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: pymongo/client_session.py [hunk 1]
```diff
@@ -952,6 +952,8 @@ def _update_read_concern(self, cmd, sock_info):
             rc['level'] = 'snapshot'
             if self._snapshot_time is not None:
                 rc['atClusterTime'] = self._snapshot_time
+    def __copy__(self):
```
Comment (ShaneHarvey) [inline pymongo/client_session.py:955]: There should be a blank line before this method.

File: test/test_session.py [hunk 2]
```diff
@@ -715,7 +715,10 @@ def test_snapshot_incompatible_with_causal_consistency(self):
             with self.client.start_session(causal_consistency=True,
                                            snapshot=True):
                 pass
-
+    def test_session_not_copyable(self):
```
Comment (ShaneHarvey) [inline test/test_session.py:718]: There should be a blank line before this method.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_session.py (hunk 2 above)
Comment (juliusgeo) [reply test/test_session.py:718]: Done.

File: pymongo/client_session.py (hunk 1 above)
Comment (juliusgeo) [reply pymongo/client_session.py:955]: Done.

Comment (ShaneHarvey) [reply pymongo/client_session.py:955]: I don't see any changes. Did you forget to push?

Comment (juliusgeo) [reply pymongo/client_session.py:956]: Yeah I git stashed them by accident. Restored it and pushed it now.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**